### PR TITLE
Update to mariner22.04 for mariner build

### PR DESCRIPTION
**Pull request by yophilav**
Update to mariner22.04 for mariner build

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

**Comment by damonbarry** [comment]
In the release branch PR, you updated Cargo.lock too. Do we need to update Cargo.lock to point to Azure/iot-identity-service@099596e116fab84ec5a42ae3efc2ce07d35e3276?

**Comment by yophilav** [comment]
> In the release branch PR, you updated Cargo.lock too. Do we need to update Cargo.lock to point to [Azure/iot-identity-service@099596e](https://github.com/Azure/iot-identity-service/commit/099596e116fab84ec5a42ae3efc2ce07d35e3276)?

I guess we can pin it to the latest main of IIS.

**Review by damonbarry — DISMISSED** [review]

File: edgelet/iotedge/src/lib.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #![deny(rust_2018_idioms, warnings)]
 #![deny(clippy::all, clippy::pedantic)]
 #![allow(
+    clippy::box_default,
```
Comment (damonbarry) [inline edgelet/iotedge/src/lib.rs:6]: Was this somehow required when you moved the build containers to Ubuntu 22.04?

Comment (yophilav) [reply edgelet/iotedge/src/lib.rs]: Nope it isn't required. Removing the rule...

**Review by damonbarry — APPROVED** [review]

**Pull request merged**